### PR TITLE
Control the fingers and the walking through the OpenXR joystick

### PR DESCRIPTION
**Pull request by S-Dafarra**
This PR takes some code of https://github.com/robotology/walking-teleoperation/tree/refactoring_and_htc

The focus is only to control the walking and the fingers. There is some code duplication with the `OculusModule`, but the plan in the future will be to deprecate that module.

### CODE REVIEW

**Comment by S-Dafarra** [comment]
Updated the license after #132 

The PR is ready for review @GiulioRomualdi

**Comment by S-Dafarra** [comment]
Friendly ping @GiulioRomualdi

**Comment by S-Dafarra** [comment]
> Friendly ping @GiulioRomualdi

Another friendly ping. This would allow moving forward in #133

File: modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp [hunk 1]
```diff
@@ -0,0 +1,1384 @@
...
+    // prepare the remotecontrolboards
+    yarp::os::Bottle remoteControlBoards;
+    remoteControlBoards.clear();
+    yarp::os::Bottle& remoteControlBoardsList = remoteControlBoards.addList();
+    for (auto& iCubPart : iCubParts)
+        remoteControlBoardsList.addString("/" + robot + "/" + iCubPart);
+
+    options.put("remoteControlBoards", remoteControlBoards.get(0));
+    options.put("localPortPrefix", "/" + name + "/remoteControlBoard");
+    yarp::os::Property& remoteControlBoardsOpts = options.addGroup("REMOTE_CONTROLBOARD_OPTIONS");
```
Comment (GiulioRomualdi) [inline modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp:189]: Not now but we may think of using blf in the future

File: modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp [hunk 2]
```diff
@@ -0,0 +1,1384 @@
...
+    if (desiredPosition.size() != m_actuatedDOFs)
+    {
+        yError() << "[RobotControlHelper::setDirectPositionReferences] Dimension mismatch between "
+                    "desired position vector and the number of controlled joints.";
+        return false;
+    }
+
+    // convert radiant to degree
+    for (int i = 0; i < m_actuatedDOFs; i++)
+        m_desiredJointValue(i) = iDynTree::rad2deg(desiredPosition(i));
```
Comment (GiulioRomualdi) [inline modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp:299]:
```suggestion
    for (int i = 0; i < m_actuatedDOFs; i++)
    {
        m_desiredJointValue(i) = iDynTree::rad2deg(desiredPosition(i));
    }
```

File: modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp [hunk 3]
```diff
@@ -0,0 +1,1384 @@
...
+    if (desiredVelocity.size() != m_actuatedDOFs)
+    {
+        yError() << "[RobotControlHelper::setVelocityReferences] Dimension mismatch between "
+                    "desired velocity vector and the number of controlled joints.";
+        return false;
+    }
+
+    // convert radiant/s  to degree/s
+    for (int i = 0; i < m_actuatedDOFs; i++)
+        m_desiredJointValue(i) = iDynTree::rad2deg(desiredVelocity(i));
```
Comment (GiulioRomualdi) [inline modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp:324]:
```suggestion
    for (int i = 0; i < m_actuatedDOFs; i++)
    {
        m_desiredJointValue(i) = iDynTree::rad2deg(desiredVelocity(i));
    }
```

File: modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp [hunk 4]
```diff
@@ -0,0 +1,1384 @@
...
+
+    // since the velocity interface use a minimum jerk trajectory a very high acceleration is set in
+    // order to use it as velocity "direct" interface
+    yarp::sig::Vector dummy(m_actuatedDOFs, std::numeric_limits<double>::max());
```
Comment (GiulioRomualdi) [inline modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp:328]: We may store this as an attribute of the class

**Review by GiulioRomualdi — APPROVED** [review]
Some minors comment but you can merge it as it is

File: modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp (hunk 4 above)
Comment (S-Dafarra) [reply modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp]: Good point! Done in https://github.com/robotology/walking-teleoperation/pull/130/commits/49b30a3bc6cd20d520ba044f28a8fc1f2fe90001

File: modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp (hunk 1 above)
Comment (S-Dafarra) [reply modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp:190]: This is indeed a good point. TBH I copied some old existing code and I did not invest too much time on it. I have opened https://github.com/robotology/walking-teleoperation/issues/134 as a reminder

**Comment by S-Dafarra** [comment]
CI passed. Merging. Thanks @GiulioRomualdi

**Pull request merged**